sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnVectorUtils.scala

```diff
@@ -38,4 +40,99 @@ object GpuColumnVectorUtils {
     columnVectors.asInstanceOf[Array[ColumnVector]]
   }
 
+  def isCaseWhenFusionSupportedType(dataType: DataType): Boolean = {
+    dataType match {
+      case BooleanType => true
+      case ByteType => true
+      case ShortType => true
+      case IntegerType => true
+      case LongType => true
+      case FloatType => true
+      case DoubleType => true
+      case StringType => true
+      case DecimalType() => true
+      case _ => false
+    }
+  }
+
+  /**
+   * Create column vector from scalars
+   * @param scalars literals
+   * @param dataType only used by creating decimal column vector,
+   *                 need to distinct decimal32, decimal64, decimal128,
+   *                 should not infer from scalars
+   * @return column vector for the specified scalars
+   */
+  def createFromScalarList(scalars: Seq[GpuScalar], dataType: DataType): CudfCV = {
+    scalars.head.dataType match {
+      case BooleanType =>
+        val booleans = scalars.map(s => s.getValue.asInstanceOf[java.lang.Boolean])
+        CudfCV.fromBoxedBooleans(booleans: _*)
+      case ByteType =>
+        val bytes = scalars.map(s => s.getValue.asInstanceOf[java.lang.Byte])
+        CudfCV.fromBoxedBytes(bytes: _*)
+      case ShortType =>
+        val shorts = scalars.map(s => s.getValue.asInstanceOf[java.lang.Short])
+        CudfCV.fromBoxedShorts(shorts: _*)
+      case IntegerType =>
+        val ints = scalars.map(s => s.getValue.asInstanceOf[java.lang.Integer])
+        CudfCV.fromBoxedInts(ints: _*)
+      case LongType =>
+        val longs = scalars.map(s => s.getValue.asInstanceOf[java.lang.Long])
+        CudfCV.fromBoxedLongs(longs: _*)
+      case FloatType =>
+        val floats = scalars.map(s => s.getValue.asInstanceOf[java.lang.Float])
+        CudfCV.fromBoxedFloats(floats: _*)
+      case DoubleType =>
+        val doubles = scalars.map(s => s.getValue.asInstanceOf[java.lang.Double])
+        CudfCV.fromBoxedDoubles(doubles: _*)
+      case StringType =>
+        val utf8Bytes = scalars.map(s => {
+          val v = s.getValue
+          if (v == null) {
+            null
+          } else {
+            v.asInstanceOf[UTF8String].getBytes
+          }
+        })
+        CudfCV.fromUTF8Strings(utf8Bytes: _*)
+      case DecimalType() =>
```

Comment:
Same problem here as above. Why is this only Decimal(10,0)?  Because `DecimalType()` is aliased to `DecimalType(10, 0)`

Reply:
Changed to : case _: DecimalType => true